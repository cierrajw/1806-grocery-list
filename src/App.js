import React, { Component } from 'react';
import Grocery from './Grocery'
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      groceries: []
    }
  }

  addGrocery = (grocery) => {
    this.setState({ groceries: [...this.state.groceries, grocery]})
  }

  render() {
    return (
      <div className="App">
        <h1>The Grocery List</h1>
        <Grocery 
          className="grocery" 
          name="bananas" 
          starred={false}
        />
      </div>
    );
  }
}

export default App;
