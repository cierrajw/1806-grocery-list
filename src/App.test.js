import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import App from './App';

describe('App', () => {
  it('should have one grocery', () => {
    // Setup && Execution
    const wrapper = shallow(<App />)

    // Expectation
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    // Setup && Execution
    const wrapper = shallow(<App />)
    const expected = { groceries: [] }

    // Expectation
    expect(wrapper.state()).toEqual(expected)
  })

  it('should add a grocery when addGrocery is called', () => {
    // Setup
    const wrapper = shallow(<App />)
    const mockGrocery = { name: 'saffron', quantity: 1 }
    const expected = [mockGrocery]


    // Execution
    wrapper.instance().addGrocery(mockGrocery)

    // Expectation
    expect(wrapper.state().groceries).toEqual(expected)
  })
})
