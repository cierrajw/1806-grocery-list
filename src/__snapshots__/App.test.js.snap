// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should have one grocery 1`] = `
<div
  className="App"
>
  <h1>
    The Grocery List
  </h1>
  <Grocery
    className="grocery"
    name="bananas"
    starred={false}
  />
</div>
`;
